ext/hash/hash.stub.php

```diff
@@ -2,48 +2,48 @@
 
 /** @generate-function-entries */
 
-function hash(string $algo, string $data, bool $raw_output = false): string|false {}
+function hash(string $algorithm, string $data, bool $binary_output = false): string|false {}
 
-function hash_file(string $algo, string $filename, bool $raw_output = false): string|false {}
+function hash_file(string $algorithm, string $filename, bool $binary_output = false): string|false {}
 
-function hash_hmac(string $algo, string $data, string $key, bool $raw_output = false): string|false {}
+function hash_hmac(string $algorithm, string $data, string $key, bool $binary_output = false): string|false {}
 
-function hash_hmac_file(string $algo, string $data, string $key, bool $raw_output = false): string|false {}
+function hash_hmac_file(string $algorithm, string $data, string $key, bool $binary_output = false): string|false {}
 
-function hash_init(string $algo, int $options = 0, string $key = ""): HashContext {}
+function hash_init(string $algorithm, int $options = 0, string $key = ""): HashContext {}
 
 function hash_update(HashContext $context, string $data): bool {}
 
 /** @param resource $handle */
 function hash_update_stream(HashContext $context, $handle, int $length = -1): int {}
 
-/** @param resource $stream_context */
+/** @param resource|null $stream_context */
 function hash_update_file(HashContext $context, string $filename, $stream_context = null): bool {}
 
-function hash_final(HashContext $context, bool $raw_output = false): string {}
+function hash_final(HashContext $context, bool $binary_output = false): string {}
 
 function hash_copy(HashContext $context): HashContext {}
 
 function hash_algos(): array {}
 
 function hash_hmac_algos(): array {}
 
-function hash_pbkdf2(string $algo, string $password, string $salt, int $iterations, int $length = 0, bool $raw_output = false): string {}
+function hash_pbkdf2(string $algorithm, string $password, string $salt, int $iterations, int $length = 0, bool $binary_output = false): string {}
 
 function hash_equals(string $known_string, string $user_string): bool {}
 
-function hash_hkdf(string $algo, string $ikm, int $length = 0, string $info = "", string $salt = ""): string {}
+function hash_hkdf(string $algorithm, string $key, int $length = 0, string $info = "", string $salt = ""): string {}
 
 #ifdef PHP_MHASH_BC
-function mhash_get_block_size(int $hash): int|false {}
+function mhash_get_block_size(int $algorithm): int|false {}
 
-function mhash_get_hash_name(int $hash): string|false {}
+function mhash_get_hash_name(int $algorithm): string|false {}
 
-function mhash_keygen_s2k(int $hash, string $input_password, string $salt, int $bytes): string|false {}
+function mhash_keygen_s2k(int $algorithm, string $password, string $salt, int $bytes): string|false {}
```

Comment:
```suggestion
function mhash_keygen_s2k(int $algorithm, string $password, string $salt, int $length): string|false {}
```
To be consistent with $length parameters on other key derivation functions (pbkdf2, hkdf)

Reply:
nice catch!